lib/src/pubspec.dart

```diff
@@ -635,3 +666,121 @@ class Pubspec extends PubspecBase {
 /// This is slightly more complicated than just checking if the scheme is
 /// 'file', since relative URIs also refer to the filesystem on the VM.
 bool _isFileUri(Uri uri) => uri.scheme == 'file' || uri.scheme == '';
+
+/// The parsed contents of a pubspec overrides file.
+///
+/// A pubspec overrides file overrides aspects of a pubspec file and is only
+/// taken into account for a root package.
+///
+/// Overrides are stored in a `pubspec_overrides.yaml` file next to the
+/// `pubspec.yaml` file that is being overridden.
+///
+/// A [PubspecOverrides] needs to be loaded after the corresponding [Pubspec],
+/// since it requires the name and [LanguageVersion] of the package to be
+/// able to parse its own contents.
+class PubspecOverrides with _PubspecParsingMixin {
+  /// The parsed YAML contents of this overrides file.
+  final YamlMap _fields;
+
+  @override
+  String get _fileType => 'pubspec overrides';
+
+  @override
+  final String? _packageName;
+
+  @override
+  final LanguageVersion _languageVersion;
+
+  @override
+  Uri? get _location => _fields.span.sourceUrl;
+
+  @override
+  final SourceRegistry? _sources;
+
+  /// The overrides for [Pubspec.dependencyOverrides] in this overrides file.
+  ///
+  /// If specified, they completely replace the [Pubspec.dependencyOverrides] in
+  /// the pubspec.
+  ///
+  /// This is `null` if there is no `dependency_overrides` section in this file.
+  Map<String, PackageRange>? get dependencyOverrides {
+    if (!_fields.containsKey('dependency_overrides')) {
+      return null;
+    }
+
+    return _dependencyOverrides ??= _parseDependencies(
+        'dependency_overrides', _fields.nodes['dependency_overrides']);
+  }
+
+  Map<String, PackageRange>? _dependencyOverrides;
+
+  /// Loads the pubspec overrides for a package located in [packageDir].
+  factory PubspecOverrides.load(
+    String packageDir,
+    SourceRegistry sources,
+    String? packageName,
+    LanguageVersion languageVersion,
+  ) {
+    var pubspecOverridesPath = path.join(packageDir, 'pubspec_overrides.yaml');
+    var pubspecOverridesUri = path.toUri(pubspecOverridesPath);
+
+    return PubspecOverrides.parse(readTextFile(pubspecOverridesPath), sources,
+        packageName, languageVersion,
+        location: pubspecOverridesUri);
+  }
+
+  PubspecOverrides({
+    String? packageName,
+    required LanguageVersion languageVersion,
+    Iterable<PackageRange>? dependencyOverrides,
+    Map? fields,
+    SourceRegistry? sources,
+  })  : _packageName = packageName,
+        _languageVersion = languageVersion,
+        _dependencyOverrides = dependencyOverrides == null
+            ? null
+            : Map.fromIterable(dependencyOverrides, key: (range) => range.name),
+        _fields = fields == null ? YamlMap() : YamlMap.wrap(fields),
+        _sources = sources;
```

Comment:
Is this used anywhere?

Reply:
I think I added it for testing but didn't end up using it, so no. I'll remove it.

Reply:
Done